### PR TITLE
complement the 3D UI API

### PR DESCRIPTION
**Pull request by JoneLau**

### CODE REVIEW

File: cocos/core/components/ui-base/canvas-component.ts [hunk 1]
```diff
@@ -71,6 +75,13 @@ const RenderMode = Enum({
 @executeInEditMode
 @disallowMultiple
 export class CanvasComponent extends Component {
+    /**
+     * @en
+     * The clearing flags of builtin camera.
```
Comment (slackmoehrle) [inline cocos/core/components/ui-base/canvas-component.ts]: The flags to clear the built in camera.

File: cocos/core/components/ui-base/canvas-component.ts [hunk 2]
```diff
@@ -86,10 +97,17 @@ export class CanvasComponent extends Component {
         }
     }
 
+    /**
+     * @en
+     * The color clearing value of builtin camera.
```
Comment (slackmoehrle) [inline cocos/core/components/ui-base/canvas-component.ts]: The color clearing value of the builtin camera.

File: cocos/core/components/ui-base/canvas-component.ts [hunk 3]
```diff
@@ -119,8 +150,13 @@ export class CanvasComponent extends Component {
     }
 
     /**
+     * @en
+     * Camera render priority.
+     * When you choose the RenderModel is INTERSPERSE, specifies the render order with other cameras.
```
Comment (slackmoehrle) [inline cocos/core/components/ui-base/canvas-component.ts]: When you choose the RenderModel of INTERSPERSE

File: cocos/core/components/ui-base/canvas-component.ts [hunk 4]
```diff
@@ -119,8 +150,13 @@ export class CanvasComponent extends Component {
...
+     * When you choose the RenderModel is OVERLAY, specifies sorting with the rest of the Canvas.
```
Comment (slackmoehrle) [inline cocos/core/components/ui-base/canvas-component.ts]: When you choose the RenderModel of OVERLAY

File: cocos/core/components/ui-base/ui-render-component.ts [hunk 5]
```diff
@@ -48,23 +48,35 @@ import { UIComponent } from './ui-component';
 ccenum(GFXBlendFactor);
 
 /**
+ * @en
+ * The shader property type of the material after the instance.
```
Comment (slackmoehrle) [inline cocos/core/components/ui-base/ui-render-component.ts]: after the instance. --> after instantiation.

what do you think @JoneLau?

File: cocos/core/components/ui-base/ui-render-component.ts [hunk 6]
```diff
@@ -48,23 +48,35 @@ import { UIComponent } from './ui-component';
...
+ *
  * @zh
  * 实例后的材质的着色器属性类型。
  */
 export enum InstanceMaterialType {
     /**
+     * @en
+     * The shader only have color properties.
```
Comment (slackmoehrle) [inline cocos/core/components/ui-base/ui-render-component.ts]: have -> has

File: cocos/core/components/ui-base/ui-render-component.ts [hunk 7]
```diff
@@ -48,23 +48,35 @@ import { UIComponent } from './ui-component';
...
+     *
      * @zh
      * 着色器只带颜色属性。
      */
     ADDCOLOR = 0,
 
     /**
+     * @en
+     * The shader have color and texture properties.
```
Comment (slackmoehrle) [inline cocos/core/components/ui-base/ui-render-component.ts]: have -> has

File: cocos/core/components/ui-base/ui-render-component.ts [hunk 8]
```diff
@@ -48,23 +48,35 @@ import { UIComponent } from './ui-component';
...
+     *
      * @zh
      * 着色器带颜色和贴图属性。
      */
     ADDCOLORANDTEXTURE = 1,
 
     /**
+     * @en
+     * The shader have color and texture properties and use grayscale mode.
```
Comment (slackmoehrle) [inline cocos/core/components/ui-base/ui-render-component.ts]: have -> has
use --> uses

File: cocos/core/components/ui-base/ui-transform-component.ts [hunk 9]
```diff
@@ -251,6 +263,11 @@ export class UITransformComponent extends Component {
     }
 
     /**
+     * @en
+     * Sets the untransformed size of the node.<br/>
+     * The contentSize remains the same no matter the node is scaled or rotated.<br/>
```
Comment (slackmoehrle) [inline cocos/core/components/ui-base/ui-transform-component.ts]: no matter -> no matter if

File: cocos/core/components/ui-base/ui-transform-component.ts [hunk 10]
```diff
@@ -251,6 +263,11 @@ export class UITransformComponent extends Component {
...
+     * All nodes has a size. Layer and Scene has the same size of the screen.
```
Comment (slackmoehrle) [inline cocos/core/components/ui-base/ui-transform-component.ts]: has -> have (both places in this sentence.

File: cocos/ui/components/graphics-component.ts [hunk 11]
```diff
@@ -375,6 +414,9 @@ export class GraphicsComponent extends UIRenderComponent {
     }
 
     /**
+     * @en
+     * Adds an circle to the path.
```
Comment (slackmoehrle) [inline cocos/ui/components/graphics-component.ts]: an -> a

File: cocos/ui/components/graphics-component.ts [hunk 12]
```diff
@@ -391,6 +433,9 @@ export class GraphicsComponent extends UIRenderComponent {
     }
 
     /**
+     * @en
+     * Adds an rectangle to the path.
```
Comment (slackmoehrle) [inline cocos/ui/components/graphics-component.ts]: an -> a

File: cocos/ui/components/graphics-component.ts [hunk 13]
```diff
@@ -408,6 +453,9 @@ export class GraphicsComponent extends UIRenderComponent {
     }
 
     /**
+     * @en
+     * Adds an round corner rectangle to the path.
```
Comment (slackmoehrle) [inline cocos/ui/components/graphics-component.ts]: an -> a

File: cocos/ui/components/layout-component.ts [hunk 14]
```diff
@@ -39,110 +39,126 @@ import { TransformBit } from '../../core/scene-graph/node-enum';
...
- * 缩放模式。
+ * @en Enum for Layout Resize Mode.
+ *
+ * @zh 缩放模式。
  */
 enum ResizeMode {
     /**
-     * @zh
-     * 不做任何缩放。
+     * @en Don't do any scale.
```
Comment (slackmoehrle) [inline cocos/ui/components/layout-component.ts]: Don't scale

File: cocos/ui/components/widget-component.ts [hunk 15]
```diff
@@ -464,6 +532,10 @@ export class WidgetComponent extends Component {
     }
 
     /**
+     * @en
+     * Vertical aligns the midpoint offset value,
```
Comment (slackmoehrle) [inline cocos/ui/components/widget-component.ts]: Vertical -> Vertically

File: cocos/ui/components/widget-component.ts [hunk 16]
```diff
@@ -440,6 +504,10 @@ export class WidgetComponent extends Component {
     }
 
     /**
+     * @en
+     * Horizontal aligns the midpoint offset value,
```
Comment (slackmoehrle) [inline cocos/ui/components/widget-component.ts]: Horizontal -> Horizontally

File: cocos/ui/components/widget-component.ts [hunk 17]
```diff
@@ -331,6 +375,10 @@ export class WidgetComponent extends Component {
     }
 
     /**
+     * @en
+     * Whether the stretched vertically, when enable the left and right alignment will be stretched vertically,
```
Comment (slackmoehrle) [inline cocos/ui/components/widget-component.ts]: Whether the stretched vertically, when enable the left and right alignment will be stretched vertically,
-->
Whether to stretch vertically, when enabled the left and right alignment will be stretched vertically,

File: cocos/ui/components/widget-component.ts [hunk 18]
```diff
@@ -320,6 +360,10 @@ export class WidgetComponent extends Component {
     }
 
     /**
+     * @en
+     * Whether the stretched horizontally, when enable the left and right alignment will be stretched horizontally,
```
Comment (slackmoehrle) [inline cocos/ui/components/widget-component.ts]: Whether the stretched horizontally, when enable the left and right alignment will be stretched horizontally,
-->
Whether to stretch horizontally, when enabled the left and right alignment will be stretched horizontally,

File: cocos/ui/components/widget-component.ts [hunk 19]
```diff
@@ -299,6 +336,9 @@ export class WidgetComponent extends Component {
     }
 
     /**
+     * @en
+     * Whether to align the horizontal.
```
Comment (slackmoehrle) [inline cocos/ui/components/widget-component.ts]: Whether to align the horizontal.
->
Whether to align horizontally.

File: cocos/ui/components/widget-component.ts [hunk 20]
```diff
@@ -277,6 +311,9 @@ export class WidgetComponent extends Component {
     }
 
     /**
+     * @en
+     * Whether to align the vertical.
```
Comment (slackmoehrle) [inline cocos/ui/components/widget-component.ts]: Whether to align the vertical.
->
Whether to align vertically.

File: cocos/ui/components/widget-component.ts [hunk 21]
```diff
@@ -262,6 +293,9 @@ export class WidgetComponent extends Component {
     }
 
     /**
+     * @en
+     * Whether to align the right.
```
Comment (slackmoehrle) [inline cocos/ui/components/widget-component.ts]: Whether to align the right.
->
Whether to align to the right.

File: cocos/ui/components/widget-component.ts [hunk 22]
```diff
@@ -247,6 +275,9 @@ export class WidgetComponent extends Component {
     }
 
     /**
+     * @en
+     * Whether to align the left.
```
Comment (slackmoehrle) [inline cocos/ui/components/widget-component.ts]: Whether to align the left.
->
Whether to align to the left.

File: cocos/ui/components/widget-component.ts [hunk 23]
```diff
@@ -232,6 +257,9 @@ export class WidgetComponent extends Component {
     }
 
     /**
+     * @en
+     * Whether to align the bottom.
```
Comment (slackmoehrle) [inline cocos/ui/components/widget-component.ts]: Whether to align the bottom.
->
Whether to align to the bottom.

File: cocos/ui/components/widget-component.ts [hunk 24]
```diff
@@ -217,6 +239,9 @@ export class WidgetComponent extends Component {
     }
 
     /**
+     * @en
+     * Whether to align the top.
```
Comment (slackmoehrle) [inline cocos/ui/components/widget-component.ts]: Whether to align the top.
->
Whether to align to the top.

File: cocos/ui/components/widget-component.ts [hunk 25]
```diff
@@ -117,57 +120,67 @@ export enum AlignMode {
...
-     * 横向对齐。
+     * @en Alignment horizontal.
+     *
+     * @zh 横向对齐。
      */
     HORIZONTAL = LEFT | CENTER | RIGHT,
     /**
-     * @zh
-     * 纵向对齐。
+     * @en Alignment vertical.
```
Comment (slackmoehrle) [inline cocos/ui/components/widget-component.ts]: Align

File: cocos/ui/components/widget-component.ts [hunk 26]
```diff
@@ -117,57 +120,67 @@ export enum AlignMode {
...
-     * 右边对齐。
+     * @en Alignment right.
+     *
+     * @zh 右边对齐。
      */
     RIGHT = 1 << 5,
     /**
-     * @zh
-     * 横向对齐。
+     * @en Alignment horizontal.
```
Comment (slackmoehrle) [inline cocos/ui/components/widget-component.ts]: Align

File: cocos/ui/components/widget-component.ts [hunk 27]
```diff
@@ -117,57 +120,67 @@ export enum AlignMode {
...
-     * 横向中心对齐。
+     * @en Alignment center.
+     *
+     * @zh 横向中心对齐。
      */
     CENTER = 1 << 4,
     /**
-     * @zh
-     * 右边对齐。
+     * @en Alignment right.
```
Comment (slackmoehrle) [inline cocos/ui/components/widget-component.ts]: Align

File: cocos/ui/components/widget-component.ts [hunk 28]
```diff
@@ -117,57 +120,67 @@ export enum AlignMode {
...
-     * 左边对齐。
+     * @en Alignment left.
+     *
+     * @zh 左边对齐。
      */
     LEFT = 1 << 3,
     /**
-     * @zh
-     * 横向中心对齐。
+     * @en Alignment center.
```
Comment (slackmoehrle) [inline cocos/ui/components/widget-component.ts]: Align

File: cocos/ui/components/widget-component.ts [hunk 29]
```diff
@@ -117,57 +120,67 @@ export enum AlignMode {
...
-     * 下边对齐。
+     * @en Alignment bottom.
+     *
+     * @zh 下边对齐。
      */
     BOT = 1 << 2,
     /**
-     * @zh
-     * 左边对齐。
+     * @en Alignment left.
```
Comment (slackmoehrle) [inline cocos/ui/components/widget-component.ts]: Align

File: cocos/ui/components/widget-component.ts [hunk 30]
```diff
@@ -117,57 +120,67 @@ export enum AlignMode {
...
-     * 垂直中心对齐。
+     * @en Alignment middle.
+     *
+     * @zh 垂直中心对齐。
      */
     MID = 1 << 1,
     /**
-     * @zh
-     * 下边对齐。
+     * @en Alignment bottom.
```
Comment (slackmoehrle) [inline cocos/ui/components/widget-component.ts]: Align

File: cocos/ui/components/widget-component.ts [hunk 31]
```diff
@@ -117,57 +120,67 @@ export enum AlignMode {
...
-     * 上边对齐。
+     * @en Alignment top.
+     *
+     * @zh 上边对齐。
      */
     TOP = 1 << 0,
     /**
-     * @zh
-     * 垂直中心对齐。
+     * @en Alignment middle.
```
Comment (slackmoehrle) [inline cocos/ui/components/widget-component.ts]: Align

File: cocos/ui/components/widget-component.ts [hunk 32]
```diff
@@ -117,57 +120,67 @@ export enum AlignMode {
...
- * Widget 的对齐标志，表示 Widget 选择对齐状态。
+ * @en Enum for Widget's alignment flag, indicating when the widget select alignment.
+ *
+ * @zh Widget 的对齐标志，表示 Widget 选择对齐状态。
  */
 export enum AlignFlags {
     /**
-     * @zh
-     * 上边对齐。
+     * @en Alignment top.
```
Comment (slackmoehrle) [inline cocos/ui/components/widget-component.ts]: Align

**Review by slackmoehrle — CHANGES_REQUESTED** [review]
See notes for each change.

File: cocos/core/components/ui-base/ui-render-component.ts (hunk 5 above)
Comment (JoneLau) [reply cocos/core/components/ui-base/ui-render-component.ts]: get it

**Comment by JoneLau** [comment]
I've updated all the comments，please review them. Thanks. @slackmoehrle

**Review by slackmoehrle — APPROVED** [review]

**Pull request merged**